### DISCUSSION TITLE
Piper read contents of clipboard

### DISCUSSION THREAD

**Discussion by colbec** [rhasspy/piper#371] [Ideas]
FWIW, piper seems to read very nicely from my Linux Gnome clipboard. Using the bash command `xclip -selection clipboard -o` this exposes the most recent copy to the clipboard, which can be written to a temporary file and then processed as normal by piper.

Using this technique I can be browsing emails in Thunderbird and if required copy a section of message text to the clipboard, and using a pre-prepared script piper will read that text for me. Saving my poor eyes.

**Comment by omega3** [comment]
This is what works for me in Plasma:
You add pattern in regex: 

```
.*
```

in add action → command:

```
echo %s | piper --model /path/to/model/en_US-libritts_r-medium.onnx -s 7 --output-raw | aplay -r 22050 -f S16_LE -t raw -
```

Enjoy!
